src/Perf/IIDOptimizer/GuidPatcher.cs

```diff
@@ -172,25 +174,35 @@ private bool PatchNonGenericTypeIID(MethodBody body, int startILIndex, TypeRefer
         private bool PatchGenericTypeIID(MethodBody body, int startILIndex, TypeReference type, int numberOfInstructionsToOverwrite)
         {
             SignaturePart rootSignaturePart = signatureGenerator.GetSignatureParts(type);
+            var methodName = $"<IIDData>{type.FullName}";
+            MethodReference guidDataMethodReference;
+            if (methodCache.ContainsKey(methodName))
+            {
+                guidDataMethodReference = methodCache[methodName];
+            }
+            else

```

Comment:
You can replace this with `if (!methodCache.TryGetValue(methodName, ... )) {`

Reply:
Done.